torch/_dynamo/variables/builder.py

```diff
@@ -1511,6 +1511,14 @@ def mark_static_input(self, value: torch.Tensor, guard: bool):
     def wrap_module(self, value: torch.nn.Module):
         from ..eval_frame import OptimizedModule
 
+        try:
+            params = list(value.parameters(recurse=False))
```

Comment:
Maybe should also handle tied weights?

Reply:
Good question - I agree this will overcount with tied weights. Do you have any thoughts on how to fix that?

I guess we could store like a cache of "already counted" tensor paramaters, and if we've already counted it once, not include it again.

Reply:
> I guess we could store like a cache of "already counted" tensor paramaters, and if we've already counted it once, not include it again.

I think this can be achieved by having a weakref set.

Reply:
Done.